src/librustc_typeck/variance/constraints.rs

```diff
@@ -144,14 +143,13 @@ impl<'a, 'tcx> ConstraintContext<'a, 'tcx> {
     fn find_binding_for_lifetime(&self, param_id: ast::NodeId) -> ast::NodeId {
         let tcx = self.terms_cx.tcx;
         assert!(is_lifetime(&tcx.map, param_id));
-        match tcx.named_region_map.defs.get(&param_id) {
-            Some(&rl::DefEarlyBoundRegion(_, _, lifetime_decl_id))
-                => lifetime_decl_id,
+        match tcx.named_region_map.get(&param_id) {
```

Comment:
@srinivasreddy could you also try suing master please? Hopefully this is a bug which has been fixed already